mhctools/netmhcii_pan.py

```diff
@@ -91,3 +98,125 @@ def prepare_allele_name(self, allele_name):
                 beta.gene,
                 beta.allele_family,
                 beta.allele_code)
+
+
+# TODO: rename this class to NetMHCIIpan3, turn this into a wrapper that figures out which version
+# of NetMHCIIpan is installed and use the appropriate wrapper
+class NetMHCIIpan3(NetMHCIIpanBase):
+    def __init__(
+            self,
+            alleles,
+            default_peptide_lengths=[15, 16, 17, 18, 19, 20],
+            program_name="netMHCIIpan",
+            process_limit=-1,
+            extra_flags=[]):
+        NetMHCIIpanBase.__init__(
+            self,
+            alleles=alleles,
+            parse_output_fn=parse_netmhciipan_stdout,
+            default_peptide_lengths=default_peptide_lengths,
+            program_name=program_name,
+            process_limit=process_limit,
+            extra_flags=extra_flags)
+
+
+class NetMHCIIpan4(NetMHCIIpanBase):
+    def __init__(
+            self,
+            alleles,
+            default_peptide_lengths=[15, 16, 17, 18, 19, 20],
+            program_name="netMHCIIpan",
+            process_limit=-1,
+            mode="elution_score",
+            extra_flags=[]):
+        """
+        Wrapper for NetMHCIIpan 4.0, using a different parser.
+        """
+
+        if mode not in ['binding_affinity', 'elution_score', 'all_output']:
+            raise ValueError("Unsupported mode", mode)
+
+        # Default to including binding affinity data (-BA flag), though the score and %rank will
```

Comment:
The "default to including binding affinity data (-BA flag)" is confusing to me - as written it looks like "-BA" will always be passed (i.e. it's not a default, it's fixed)?

Reply:
Updated the comment